skbio/io/_iosources.py

```diff
@@ -161,6 +167,13 @@ def get_reader(self):
     def get_writer(self):
         return self.file
 
+    def can_seek(self):
+        return self.get_reader().seekable()
+
+    def get_seekable(self):
+        # Convert the stream to a StringIO, which is seek compatible
+        return io.StringIO(self.get_reader().read())
```

Comment:
I think this assumption would require reading the entirety of `stdin`, which could exceed system memory?

Reply:
Yeah, that's true. I'm currently looking at other solutions, so I'll close this pull request.

Reply:
@RaeedA You don't have to close this PR. You can update the code and commit, which will update this PR. This is part of normal code review process.